pkg/utils/release/controller.go

```diff
@@ -39,50 +42,52 @@ func (s *SuiteController) CreateApplicationSnapshot(name string, namespace strin
 }
 
 // CreateRelease creates a new Release using the given parameters.
-func (s *SuiteController) CreateRelease(name, namespace, snapshot, sourceReleaseLink string) (*v1alpha1.Release, error) {
-	release := &v1alpha1.Release{
+func (s *SuiteController) CreateRelease(name, source_namespace, snapshot, sourceReleasePlan string) (*appstudiov1alpha1.Release, error) {
+	release := &appstudiov1alpha1.Release{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      name,
-			Namespace: namespace,
+			Namespace: source_namespace,
 		},
-		Spec: v1alpha1.ReleaseSpec{
+		Spec: appstudiov1alpha1.ReleaseSpec{
 			ApplicationSnapshot: snapshot,
-			ReleaseLink:         sourceReleaseLink,
+			ReleasePlan:         sourceReleasePlan,
 		},
 	}
 
 	return release, s.KubeRest().Create(context.TODO(), release)
 }
 
-// CreateReleaseLink creates a new ReleaseLink using the given parameters.
-func (s *SuiteController) CreateReleaseLink(name, namespace, application, targetNamespace, releaseStrategy string) (*v1alpha1.ReleaseLink, error) {
-	releaseLink := &v1alpha1.ReleaseLink{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      name,
-			Namespace: namespace,
-		},
-		Spec: v1alpha1.ReleaseLinkSpec{
-			DisplayName:     name,
-			Application:     application,
-			Target:          targetNamespace,
-			ReleaseStrategy: releaseStrategy,
-		},
-	}
+// GetRelease returns the release with the given name in the given namespace.
+func (s *SuiteController) GetRelease(releaseName, releaseNamespace string) (*v1alpha1.Release, error) {
+	release := &v1alpha1.Release{}
 
-	return releaseLink, s.KubeRest().Create(context.TODO(), releaseLink)
+	err := s.KubeRest().Get(context.TODO(), types.NamespacedName{
+		Name:      releaseName,
+		Namespace: releaseNamespace,
+	}, release)
+
+	return release, err
 }
 
 // CreateReleaseStrategy creates a new ReleaseStrategy using the given parameters.
-func (s *SuiteController) CreateReleaseStrategy(name, namespace, pipelineName, bundle string, policy string) (*v1alpha1.ReleaseStrategy, error) {
-	releaseStrategy := &v1alpha1.ReleaseStrategy{
+func (s *SuiteController) CreateReleaseStrategy(name, target_namespace, pipelineName, bundle string, policy string, service_account string) (*appstudiov1alpha1.ReleaseStrategy, error) {
+	releaseStrategy := &appstudiov1alpha1.ReleaseStrategy{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      name,
-			Namespace: namespace,
+			Namespace: target_namespace,
 		},
-		Spec: v1alpha1.ReleaseStrategySpec{
+		Spec: appstudiov1alpha1.ReleaseStrategySpec{
 			Pipeline: pipelineName,
 			Bundle:   bundle,
 			Policy:   policy,
+			Params: []appstudiov1alpha1.Params{
```

Comment:
This test and all the current e2e-tests are aligned with m7 of hacbs-demos
https://github.com/scoheb/hacbs-demos/blob/main/m7/release/managed-workspace/release_strategy.yaml

Reply:
Yes but the m7 demo cared about what the release pipelinerun actually did? Does this? If it does, I don't see where any of it is checked. So I don't see why we need to add this in

Reply:
Still an open question

Reply:
I just adapted the release Strategy to the M7 demo
As for what release pipelinerun actually did , that Kasem is handling in re-enabling the happy-path tests (HACBS-1108)

Reply:
We are checking as before the regular checks:
return pipelineRun.HasStarted() && pipelineRun.IsDone() && pipelineRun.Status.GetCondition(apis.ConditionSucceeded).IsTrue()

Reply:
Even if the m7 demo has it, it sounds like you don't need it here, but from your reply maybe it will make life easier for Kasem so I guess we can keep it